pkg/boot/uefi/uefi.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uefi
+
+import (
+	"bytes"
+	"debug/pe"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math"
+	"os"
+
+	"github.com/u-root/u-root/pkg/acpi"
+	"github.com/u-root/u-root/pkg/boot"
+	"github.com/u-root/u-root/pkg/boot/kexec"
+	"github.com/u-root/u-root/pkg/smbios"
+)
+
+const fvEntryImageOffset int64 = 0xA0
+
+var bootExecute = boot.Execute
+var kexecLoad = kexec.Load
+var kexecParseMemoryMap = kexec.ParseMemoryMap
+var getRSDP = acpi.GetRSDP
+
+func getSMBIOSBaseInternal() (int64, int64, error) {
+	base, size, err := smbios.GetSMBIOSBaseEFI()
+	if err != nil {
+		base, size, err = smbios.GetSMBIOSBaseLegacy()
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+	return base, size, nil
+}
+
+var getSMBIOSBase = getSMBIOSBaseInternal
+
+// Serial port setting in Linuxboot.h
+const (
+	SerialPortTypeIO   = 1
+	SerialPortTypeMMIO = 2
+)
+
+// SerialPortConfig defines debug port configuration
+type SerialPortConfig struct {
```

Comment:
i know it's tedious, but can you mention where each of these structs come from? (are these technically our HOBs?) something more specific than "Linuxboot.h" because that could be referring to anything

Reply:
I updated the comments. PTAL